### ISSUE TITLE
openFile().read() allocates way too much

### ISSUE THREAD

**Issue by WebFreak001** [vibe-d/vibe-core#122]
```
bytes allocated, allocations, type, function, file:line
        1769952              18437    std.parallelism.Task!(taskFun, ThreadedFileEventDriver!(PosixEventDriverEvents!(EpollEventLoop, PosixEventDriverSockets!(EpollEventLoop))), FileFD, ulong, ubyte[]).Task std.parallelism.task!(taskFun, ThreadedFileEventDriver!(PosixEventDriverEvents!(EpollEventLoop, PosixEventDriverSockets!(EpollEventLoop))), FileFD, ulong, ubyte[]).task /usr/include/dlang/dmd/std/parallelism.d:835
```

reading into a buffer but it still allocated nearly 2MB for a 3MB file I read in 4096 byte chunks.

I switched to `std.stdio.File(path, "rb").byChunk(buffer)` with occasional yields now which resulted in no GC calls anymore and a speedup from 400-500ms to 100ms.

basically what I did was:
```d
ubyte[4096] buffer;
auto indices = appender!(size_t[]);
indices.reserve(1024 * 16);

auto f = openFile(path);
while (!f.empty)
{
    ubyte[] chunk = buffer[0 .. min(f.leastSize, $)];
    f.read(chunk);
    process(chunk, indices);
}

foreach (index; indices)
{
    f.seek(index);
    while (!f.empty)
    {
        ubyte[] chunk = buffer[0 .. min(f.leastSize, $)];
        f.read(chunk);
        if (processBlock(chunk)) break;
    }
}
```
where `process` searched for indexes in the file to then all go through in sequence using `processBlock`, which both don't allocate. I don't know if the seek part is important here but it felt awfully slow and for some reason it just allocated so much.

**Comment by s-ludwig** [comment]
Looks like these are the `std.parallelism.Task` objects. They would have to be put as plain fields into `FileInfo` instead to avoid the heap allocation.

**Comment by s-ludwig** [comment]
Should be fixed by vibe-d/eventcore#213
